src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,123 @@
-export const Pagination = () => {};
+import React from 'react';
+import cn from 'classnames';
+
+type Props = {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (page: number) => void;
+};
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  let pageCount = 0;
+  const pages: number[] = [];
+  const itemsArr: number[] = [];
+
+  for (let i = 1; i <= perPage; i += 1) {
+    const resultItem = i + perPage * (currentPage - 1);
+
+    itemsArr.push(resultItem);
+  }
+
+  for (let i = total; i > 0; i -= perPage) {
+    pageCount += 1;
+    pages.push(pageCount);
+  }
+
+  return (
+    <>
+      <ul className="pagination">
+        <li className={cn('page-item', {
+          disabled: currentPage === 1,
+        })}
+        >
+          {currentPage === 1 ? (
+            <a
+              data-cy="prevLink"
+              className="page-link"
+              href="#prev"
+              aria-disabled="true"
+            >
+              «
+            </a>
+          ) : (
+            <a
+              data-cy="prevLink"
+              className="page-link"
+              href="#prev"
+              aria-disabled="false"
+              onClick={() => {
+                onPageChange(currentPage - 1);
+              }}
+            >
+              «
+            </a>
+          )}
+        </li>
+        {pages.map(page => (
+          <li
+            className={cn('page-item', {
+              active: currentPage === page,
+            })}
+            key={page}
+          >
+            <a
+              data-cy="pageLink"
+              className="page-link"
+              href={`#${page}`}
+              onClick={() => {
+                if (currentPage !== page) {
+                  onPageChange(page);
+                }
+              }}
```

Comment:
Consider will be better if you create a helper function for this logic

Reply:
done